ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAPP-93] Get next application by caseworker work stream

## Description of change
Get next application assigns a caseworker an extradition case if they are assigned to an extradition work stream. Caseworkers that are not assigned to the extradition work stream will be assigned the next unassigned application regardless of work stream.

## Link to relevant ticket
[CRIMAPP-93](https://dsdmoj.atlassian.net/browse/CRIMAPP-93)

## Notes for reviewer
Note this only handles a caseworker with one competency (extradition)

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

**Caseworker competencies** 

<img width="1233" alt="Screenshot 2023-11-16 at 16 29 15" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/98fb35c1-a607-4538-926a-6e3c683beebe">

**Caseworker with no competencies set is assigned the next unassigned application** 

<img width="1233" alt="Screenshot 2023-11-16 at 16 28 53" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/74cc445b-4dd8-4df1-bc50-aad544f26301">
<img width="1233" alt="Screenshot 2023-11-16 at 16 28 59" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/d11e3190-4573-4f89-9064-adc39243669b">

**Caseworker with extradition work stream set is assigned the next extradition application** 

<img width="1233" alt="Screenshot 2023-11-16 at 16 28 38" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/03da69f8-7e5b-4af7-91ab-649ecc7a5b8f">
<img width="1233" alt="Screenshot 2023-11-16 at 16 28 32" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/c9e88617-6aac-4b8b-adf5-a851eae5decc">


## How to manually test the feature
Requires one supervisor user and two caseworker users (can also use one caseworker user if you want).
Also requires pulling down the datastore changes and spinning up review, apply and the datastore.

In apply, you'll need to submit an application with a first court hearing at Westminster Magistrates Court. You can also submit an application which did not have/is not having a first court hearing at Westminster Magistrates Court. 

Log in as a supervisor user in review and go to the manage competencies page, where you will set the competency of one of the caseworkers as extradition and the other caseworker can remain unset. 

Then log in as the caseworker assigned to the extradition work stream and go to assigned applications and click Get Next. You should be assigned to the application you just submitted with a first court hearing at Westminster Magistrates Court. (Or you may see an application you submitted earlier if you had an older unassigned application with a first court hearing at Westminster Magistrates Court.)

Then log in as the caseworker not assigned to the extradition work stream and go to assigned applications and click Get Next. You should not be assigned to the application you just submitted with a first court hearing at Westminster Magistrates Court. You should be assigned to the unassigned application that is next up in the queue. 

Play around with setting and unsetting the extradition competency and see if you can spot any weird behaviour

[CRIMAPP-93]: https://dsdmoj.atlassian.net/browse/CRIMAPP-93?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
